docs/madr/decisions/056-policy-matching-with-real-resources.md

```diff
@@ -0,0 +1,548 @@
+name: other-service
+labels: {}
+---
+type: Policy
+mesh: mesh-1
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+         kind: Mesh
+       conf: $conf1
+---
+type: Policy
+mesh: mesh-1
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+         kind: MeshService
+         labels:
+             a-common-label: foo
+       conf: $conf2
+---
+type: Policy
+mesh: mesh-1
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+         kind: MeshService
+         name: my-service
+       conf: $conf3
+---
+type: Policy
+mesh: mesh-1
+spec:
+  targetRef:
+    kind: MeshSubset
+    labels:
+      foo: bar
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          another-common-label: foo
+        conf: $conf4
+```
+
+Kuma CP should produce the following structure for the DPP that has the label `foo: bar`:
+
+```yaml
+resourceRules:
+  meshservice:mesh/mesh-1:name/my-service:
+    conf: merge($conf1, $conf2, $conf3)
+  meshservice:mesh/mesh-1:name/labeled-service:
+    conf: merge($conf1, $conf2, $conf4)
+  mesh:name/mesh-1:
+    conf: $conf1
+```
+
+Note that the structure doesn't contain `other-service`, but instead it contains entry for `mesh-1`.
+The code that computes confs should handle the absence of `other-service` and replace it with conf of the `mesh-1`.
+
+### Inspect API
+
+When it comes to the Inspect API, we're going to add a new field to `InspectRule`
+
+```yaml
+InspectRule:
+  type: object
+  required: [type]
+  properties:
+    type:
+      type: string
+      example: MeshRetry
+      description: the type of the policy
+    proxyRule:
+      description: a rule that affects the entire proxy
+      $ref: '#/components/schemas/ProxyRule'
+    toRules: # should be removed eventually
+      type: array 
+      description: a set of rules for the outbounds of this proxy
+      items:
+        $ref: '#/components/schemas/Rule'
+    toResourceRules: # new field
+      type: array
+      items:
+        $ref: '#/components/schemas/ResourceRule'
+    fromRules:
+      type: array
+      description: a set of rules for each inbound of this proxy
+      items:
+        $ref: '#/components/schemas/FromRule'
+    warnings:
+      type: array
+      description: a set of warnings to show in policy matching
+      example: ["Mesh is not Mtls enabled this policy will have no effect"]
+      items:
+        type: string
+Rule:
+  type: object
+  required: [matchers, conf, origin]
+  properties:
+    matchers:
+      type: array
+      items:
+        $ref: '#/components/schemas/RuleMatcher'
+    conf:
+      description: The actual conf generated
+      type: object
+      x-go-type: 'interface{}'
+    origin:
+      type: array
+      items:
+        $ref: '#/components/schemas/Meta'
+ResourceRule: # new type
+  type: object
+  required: [resourceMeta, conf, origin]
+  properties:
+    resourceMeta:
+      $ref: '#/components/schemas/Meta'
+    resourceSectionName:
+      type: string
```

Comment:
Should the sectionName be included in the resourceMeta?

Reply:
ResourceMeta identifies the entire resource, we can't put `sectionName` inside it. I could've created a new type like:
```go
type ResourceSection struct {
    ResourceMeta
    SectionName string
}
```
but the usage at this moment is scoped to ResourceRules (and Inspect API), so I don't think introducing new type makes much sense